lib/components/session/login.vue

```diff
@@ -45,24 +43,21 @@ export default {
       disabled: false
     };
   },
-  mounted() {
-    this.$refs.email.focus();
-  },
   methods: {
     logIn() {
       this.disabled = true;
       axios
         .post('/sessions', { email: this.email, password: this.password })
         .then(response => {
           const success = this.$session.set(response.data);
-          const headers = axios.defaults.headers.common;
           if (success) {
+            const headers = axios.defaults.headers.common;
             headers.Authorization = `Bearer ${this.$session.token}`;
-            // Using safeBack() rather than this.$router.go(-1) in case the user
-            // navigated directly to /login rather than being redirected there.
-            safeBack('/forms');
+            const next = this.$route.query.next;
+            const nextPattern = /^(\/|(\/[\w-]+)+)$/;
```

Comment:
?

are you trying to determine that the route is local?

Reply:
Exactly, I'm pretty sure the router will check this anyway, but I thought it didn't hurt to add a second check of the query parameter.

Reply:
hm. i think the safer way is to use the browser's own facilities:

    const link = document.createElement('a');
    link.href = this.$route.query.next;
    link.host === window.location.host;